### ISSUE TITLE
There is a conflict between ignores and languageOptions

### ISSUE THREAD

**Issue by chenglu4343** [antfu/eslint-config#594]
### Describe the bug

if I use 

```js
export default antfu({
  ignores: [
    'test.js',
  ],
  // languageOptions: {},
})
```

run command

```bash
npx eslint test.js
```

i get 

```
0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to disable file ignore settings or use "--no-warn-ignored" to suppress this warning
```

if i use
 
```js
export default antfu({
  ignores: [
    'test.js',
  ],
  languageOptions: {},
})
```

i get 

```bash
 1:1  error  Unexpected console statement  no-console
```

ignores configuration entries will not take effect

### Reproduction

https://github.com/chenglu4343/try-eslint/tree/try-ignores

### System Info

```Shell
System:
    OS: Windows 11 10.0.22631
    CPU: (12) x64 12th Gen Intel(R) Core(TM) i5-12400
    Memory: 3.03 GB / 15.78 GB
  Binaries:
    Node: 18.20.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.22 - C:\Program Files\nodejs\yarn.CMD
    npm: 10.5.0 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.15.6 - C:\Program Files\nodejs\pnpm.CMD
  Browsers:
    Edge: Chromium (127.0.2651.98)
    Internet Explorer: 11.0.22621.3527
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

**Comment by chenglu4343** [comment]
This seems to be an issue with eslint. The ignores configuration items cannot contain other keys

// eslint.config.mjs

```js
export default [
  {
    ignores: ["test.js"],
    ddd: {},
  },
];
```

This ignores configuration items cannot take effect.

So we need to do some special processing on the ignores configuration item.

https://github.com/antfu/eslint-config/blob/main/src/factory.ts#L279

**Comment by antfu** [comment]
That's how ESLint flat config is designed to work (I agree it's a bit confusing, but that's how it is). Please read the docs more carefully.

**Comment by chenglu4343** [comment]
But my project needs to configure globals options and ignores Ignores file configuration items at the same time, now I can not do it

**Comment by Dimava** [comment]
@antfu actually in this specific case I agree with OP, `ignores` from the primary config(the first one) should go into `ignores()` config
If you see a config like 
```ts
export default antfu({
  ignores: ['file.ts'],
  vue: true,
  typescript: {
    parserOptions: {
      projectService: true,
    },
  },
  gitignore: true,
  rules: {
    'roole': 'error',
  }
}, {
    ignores: ['file2.ts'],
}, {
    ignores: ['file3.ts'],
    rules: { 'override': 'error' }
})
```
you'd expect `ignores: ['file.ts'],` to actually be global, unlike `file2`/`file3` that you'd expect to work as is
Maybe if the first arg would be just `options?: OptionsConfig` rather then `options?: OptionsConfig & TypedFlatConfigItem` it'd be better? (or `options?: OptionsConfig | TypedFlatConfigItem`)
Otherwise global and local definitions in the same object cause confusion
